### DISCUSSION TITLE
trpc

### DISCUSSION THREAD

**Discussion by minox86** [ardatan/feTS#42] [General]
How would it differ from https://trpc.io/?

**Comment by m-abdelwahab** [comment]
Based on my understanding, I would say the main fundamental differences between FeTS and tRPC are:

- When using the FeTS server, you build a REST API while following the OpenAPI spec. With tRPC, you're writing procedures- remote functions you call
- FeTS server and client components are decoupled, and each one of them can be used on their own. With tRPC it doesn't make sense to use the client without the server

**Comment by AndKenneth** [comment]
I actually use both. trpc-openapi lets me publish certain trpc procedures as a rest endpiont (while still using them in the standard trpc/tanstack query way in the frontend of that app), then I can use the generated openapi spec to generate a typesafe client with fets client on a completely uncoupled app with a bit of copy paste.

I explored doing fully fets but we really love using trpc when things are tightly coupled. We could also have moved things to a monorepo and used trpc everywhere, but then we still probably need to publish an endpoint when we have something that HAS to be decoupled fully, like work by an external integrator, code in another language, or future public api use.
